src/pages/List.js

```diff
@@ -1,47 +1,17 @@
 import React, { useContext } from 'react';
-import { FirestoreCollection } from 'react-firestore';
-import Loading from '../components/Loading';
 import NavTabs from '../components/NavTabs';
-import ErrorMessage from '../components/ErrorMessage';
 import HomePageButton from '../components/HomePageButton';
 import { ListContext } from '../listContext';
+import { TokenContext } from '../tokenContext';
 
 const List = () => {
-  const { setShoppingList, shoppingList } = useContext(ListContext);
-  const getStoredToken = () => window.localStorage.getItem('token');
+  const { token } = useContext(TokenContext);
+  const { displayList } = useContext(ListContext);
 
   return (
     <>
       <HomePageButton />
```

Comment:
I might simply call the `fetchList` from `ListContext` which should take a token as an argument. Then you can get the data you need to render the list outside of `ListContext`